### PR TITLE
Improve ECDSA verify validation - 2.28 backport

### PR DESCRIPTION
**Pull request by daverodgman**
Trivial backport of #6190

### CODE REVIEW

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM. Faithful backport

**Comment by tom-cosgrove-arm** [comment]
(I will leave the `approved` off this until the main PR has two approvals)

File: library/ecdsa.c
Comment (gilles-peskine-arm) [inline library/ecdsa.c:638]: Either error code is arguably correct. In an LTS, we should err on the side of being conservative.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Rejecting the error code change in an LTS branch.

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM. Faithful backport.

**Comment by daverodgman** [comment]
Trivial rebase onto latest mbedtls-2.28 just to test Travis - no changes made

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
This is a failthful backport of https://github.com/Mbed-TLS/mbedtls/pull/6190, and has the same issues in the changelog entry.

**Review by tom-cosgrove-arm — APPROVED** [review]
Faithful backport, now matching #6190 with fixed ChangeLog entry

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**
